lib/rom/mapper.rb

```diff
@@ -56,7 +56,18 @@ def self.register_processor(processor)
     #
     # @api private
     def self.build(header = self.header, processor = :transproc)
-      new(Mapper.processors.fetch(processor).build(header), header)
+      processor = Mapper.processors.fetch(processor)
+
+      if steps.empty?
+        new(processor.build(header), header)
+      elsif attributes.any?
+        raise(MapperMisconfiguredError, "cannot mix outer attributes and steps")
+      else
+        step_transformers = steps.map(&:header).map { |step_header|
+          new(processor.build(step_header), step_header).transformers.first
```

Comment:
Yes, it would be. I cannot change this PR, because it it the @dekz who owns it. Can only update after the merge